dada2_workflow.Rmd

```diff
@@ -0,0 +1,206 @@
+```
+
+```{r dada2_2}
+path <- "~/Desktop/dada2_example/MiSeq_SOP" # CHANGE ME to the directory containing the fastq files after unzipping.
+list.files(path)
+```
+
+```{r dada2_3}
+# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
+fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
+fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
+# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
+sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
+```
+
+```{r dada2_4}
+# INSPECT READ QUALITY PROFILES
+
+plotQualityProfile(fnFs[1:2])
+```
+
+```{r dada2_5}
+plotQualityProfile(fnRs[1:2])
+```
+
+```{r dada2_6}
+# FILTER AND TRIM
+
+# Place filtered files in filtered/ subdirectory
+filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
+filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
+names(filtFs) <- sample.names
+names(filtRs) <- sample.names
+```
+
+```{r dada2_7}
+out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
+              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
+              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
+head(out)
+```
+
+```{r dada2_8}
+# LEARN THE ERROR RATES
+
+errF <- learnErrors(filtFs, multithread=TRUE)
+```
+
+```{r dada2_9}
+errR <- learnErrors(filtRs, multithread=TRUE)
+```
+
+```{r dada2_10}
+plotErrors(errF, nominalQ=TRUE)
+```
+
+```{r dada2_11}
+# SAMPLE INFERENCE
+
+dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
+```
+
+```{r dada2_12}
+dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
+```
+
+```{r dada2_13}
+dadaFs[[1]]
+```
+
+```{r dada2_14}
+# MERGE PAIRED READS
+
+mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
+# Inspect the merger data.frame from the first sample
+head(mergers[[1]])
+```
+
+```{r dada2_15}
+# CONSTRUCT SEQUENCE TABLE
+
+seqtab <- makeSequenceTable(mergers)
+dim(seqtab)
+```
+
+```{r dada2_16}
+# Inspect distribution of sequence lengths
+table(nchar(getSequences(seqtab)))
+```
+
+```{r dada2_17}
+# REMOVE CHIMERAS
+
+seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
+dim(seqtab.nochim)
+```
+
+```{r dada2_18}
+sum(seqtab.nochim)/sum(seqtab)
+```
+
+```{r dada2_19}
+# TRACK READS THROUGH THE PIPELINE
+
+getN <- function(x) sum(getUniques(x))
+track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
+# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
+colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
+rownames(track) <- sample.names
+head(track)
+```
+
+```{r dada_20}
+# ASSIGN TAXONOMY
+
+# Change path
+taxa <- assignTaxonomy(seqtab.nochim, "~/Desktop/dada2_example/tax/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
```

Comment:
no local paths. 

Also cite the DADA2 website that provides now more updated reference databases in the readily usable format (I sent the link through slack earlier, can resend if necessary)